modules/core/src/lapack.cpp

```diff
@@ -43,6 +43,15 @@
 #include "precomp.hpp"
 #include <limits>
 
+#ifdef HAVE_EIGEN
+#  if defined __GNUC__ && defined __APPLE__
+#    pragma GCC diagnostic ignored "-Wshadow"
+#  endif
```

Comment:
Why we need to suppress this warning?
Perhaps related code can be refactored.

Reply:
OK let me look into this. I see this in every single include of Eigen in opencv; I just copied it. I also don't have a MAC.